server/handlers/ui_handler.go

```diff
@@ -37,24 +39,36 @@ func getDynamicUIEndpoint(reqURL string) string {
 }
 
 // ServeUI - helps serve static files for both meshery ui and provider ui
-func ServeUI(w http.ResponseWriter, r *http.Request, reqBasePath, baseFolderPath string) {
+func (h *Handler) ServeUI(w http.ResponseWriter, r *http.Request, reqBasePath, baseFolderPath string) {
 	// if r.Method != http.MethodGet {
 	// 	http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
 	// 	return
 	// }
+
 	reqURL := r.URL.Path
 	reqURL = strings.Replace(reqURL, reqBasePath, "", 1)
 
 	var filePath strings.Builder
+	if reqURL == "/" && viper.Get("RELEASE_CHANNEL") == (models.Kanvas{}).String() {
```

Comment:
@MUzairS15 this is not ideal. I've discussed the two possible other implementations to ensure this is an area of extensibility.